plugins/snapshots/erofs/erofs_linux.go

```
@@ -235,14 +354,58 @@ func (s *snapshotter) formatLayerBlob(id string) error {
 		if err != nil {
 			return fmt.Errorf("failed to format dmverity: %w", err)
 		}
+
 		dmverityData := fmt.Sprintf("%s|%d", info.RootHash, fileinfo.Size())
 		if err := os.WriteFile(filepath.Join(s.root, "snapshots", id, ".dmverity"), []byte(dmverityData), 0644); err != nil {
 			return fmt.Errorf("failed to write dmverity root hash: %w", err)
 		}
+
+		var layerDigest string = ""
+		if snapshotInfo.Labels != nil {
+			if digest, ok := snapshotInfo.Labels[snapshotters.TargetLayerDigestLabel]; ok {
+				layerDigest = digest
+				log.L.Infof("found layer digest in labels: %s", layerDigest)
+			}
+		} else {
+			log.L.Debugf("snapshotInfo.Labels is nil")
+		}
+
+		log.L.Info("Before checking layerDigest s.signatures")
+		// If we have the layer digest and signatures map is available
+		if layerDigest != "" && s.signatures != nil {
```

What does it mean if we dont have a layerDigest at this point?

---

I believe it is similar to this [check](https://github.com/microsoft/kata-containers/blob/erofs-main/src/tardev-snapshotter/src/snapshotter.rs#L1404) in the tardev snapshotter. At this point, if we follow the same pattern as in the tardev, we should error out if we do not have the layerDigest

---

Yeah, I think we should assert that it is present rather than silently continue. This is good time to catch any of these issues early on.